src/project/config.rs

```diff
@@ -0,0 +1,301 @@
+                                            Err(err) => {
+                                                println!("{}: {}, expecting: `{}`", CONFIG_NAME, err, stringify!($ty));
+                                                Some($default)
+                                            },
+                                        };
+                                    },
+                                )*
+                            }
+                        }
+
+                        $(
+                            let $field = match $field {
+                                Some(v) => v,
+                                None => {
+                                    use ::serde::de::Error;
+                                    let err: V::Error = V::Error::missing_field(stringify!($field));
+                                    println!("{}: {}, expecting: `{}`", CONFIG_NAME, err, stringify!($ty));
+                                    $default
+                                },
+                            };
+                        )*
+
+                        Ok(
+                            $identifier {
+                                $(
+                                    $field: $field,
+                                )*
+                            }
+                        )
+                    }
+                }
+
+                deserializer.deserialize_struct(stringify!($identifier), FIELDS, IdentifierVisitor)
+            }
+        }
+
+        impl ::std::default::Default for $identifier {
+            fn default() -> Self {
+                $identifier {
+                    $(
+                        $field: $default,
+                    )*
+                }
+            }
+        }
+
+        impl $crate::project::Config for $identifier {
+            fn load<P: AsRef<::std::path::Path>>(path: P) -> Self {
+                match $identifier::direct_load(path.as_ref()) {
+                    Ok(v) => v,
+                    Err(err) => {
+                        println!("{}: {}", stringify!($identifier), err.description());
+                        $identifier::default()
+                    },
+                }
+            }
+
+            fn write<P: AsRef<::std::path::Path>>(&self, path: P) -> Result<(), $crate::project::ProjectError> {
+                use ::std::io::Write;
+
+                let result = ::serde_yaml::to_string(self);
+                let serialized = try!(result.map_err(|e| $crate::project::ProjectError::Parser(e.to_string()) ));
+                let mut file = try!(::std::fs::File::create(path));
+                try!(file.write(&serialized.into_bytes()));
+
+                Ok(())
+            }
+
+            fn direct_load<P: AsRef<::std::path::Path>>(path: P) -> Result<Self, $crate::project::ProjectError> {
+                let content = try!($crate::project::directory::Directory::load(path));
+                let parsed = ::serde_yaml::from_str::<$identifier>(&content);
```

Comment:
There was already some discussion on Gitter about supporting multiple formats and also #171 addresses this issue. Should we do it? It would probably be a simple

```rust
match path.file_extension() {
    "yml" => serde_yaml::from_str(&content),
    "toml" => toml::from_str(&content),
}
```